advanced/wallets/react-wallet-v2/src/utils/EIP5792RequestHandlerUtils.ts

```diff
@@ -0,0 +1,84 @@
+import { EIP155_CHAINS, TEIP155Chain } from '@/data/EIP155Data'
+import { EIP5792_METHODS, GetCapabilitiesResult, SendCallsParams, supportedEIP5792CapabilitiesForEOA, supportedEIP5792CapabilitiesForSCA } from '@/data/EIP5792Data'
+import { getWallet } from '@/utils/EIP155WalletUtil'
+import { formatJsonRpcError, formatJsonRpcResult } from '@json-rpc-tools/utils'
+import { SignClientTypes } from '@walletconnect/types'
+import { getSdkError } from '@walletconnect/utils'
+import { providers } from 'ethers'
+import { KernelSmartAccountLib } from '@/lib/smart-accounts/KernelSmartAccountLib'
+import SettingsStore from '@/store/SettingsStore'
+import { smartAccountWallets } from './SmartAccountUtil'
+import EIP155Lib from '@/lib/EIP155Lib'
+import { SmartAccountLib } from '@/lib/smart-accounts/SmartAccountLib'
+type RequestEventArgs = Omit<SignClientTypes.EventArguments['session_request'], 'verifyContext'>
+
+
+export async function approveEIP5792Request(requestEvent: RequestEventArgs) {
+  const { params, id } = requestEvent
+  const { chainId, request } = params
+
+  console.log(requestEvent, chainId, 'tests')
+
+  SettingsStore.setActiveChainId(chainId)
+
+  const wallet = await getWallet(params)
+
+  switch (request.method) {
+    case EIP5792_METHODS.WALLET_GET_CAPABILITIES:
+      /**
+       * If the account in the params is a EOA return supportedEIP5792CapabilitiesForEOA
+       * If the account in the params is a Smart Contract return supportedEIP5792CapabilitiesForSCA
+       */
+      if(wallet instanceof EIP155Lib)
+        return formatJsonRpcResult<GetCapabilitiesResult>(id, supportedEIP5792CapabilitiesForEOA)
+      else if(wallet instanceof SmartAccountLib || wallet instanceof KernelSmartAccountLib)
+        return formatJsonRpcResult<GetCapabilitiesResult>(id, supportedEIP5792CapabilitiesForSCA)
```

Comment:
if we return we don't need else

Reply:
We may also want  to define some (maybe even empty?) interface that will be implemented by `SmartAccountLib` and `KernelSmartAccountLib` so we can just check for that